stratum/hal/lib/common/config_monitoring_service.cc

```diff
@@ -131,6 +134,39 @@ ::util::Status ConfigMonitoringService::PushChassisConfig(
   return ::util::OkStatus();
 }
 
+namespace {
+// Helper function to determine whether a container of protobuf messages have an
+// unique name field.
+template <typename T>
+bool AreUniqueNames(const T& values) {
+  absl::flat_hash_set<std::string> unique_names;
+  for (const auto& e : values) {
+    if (e.name().empty()) continue;
+    if (!gtl::InsertIfNotPresent(&unique_names, e.name())) {
+      return false;
+    }
+  }
+  return true;
+}
+}  // namespace
+
+::util::Status ConfigMonitoringService::VerifyChassisConfig(
+    const ChassisConfig& config) {
+  // Validate the names of the components, if given.
+  CHECK_RETURN_IF_FALSE(AreUniqueNames(config.nodes()));
+  CHECK_RETURN_IF_FALSE(AreUniqueNames(config.singleton_ports()));
+  CHECK_RETURN_IF_FALSE(gtl::STLIsUnique(
+      config.singleton_ports(),
+      [](const SingletonPort& a, const SingletonPort& b) {
+        return !a.name().empty() && !b.name().empty() && a.name() == b.name();
+      }));
```

Comment:
Decide on implementation approach.

Reply:
@bocon13 @Yi-Tseng Any thoughts on these approaches?

`STLIsUnique` has O(n^2) complexity but uses no additional memory. `AreUniqueNames` uses a set but is amortized O(n).

Reply:
I think we can use `AreUniqueNames` since we won't have a huge set of port configs (up to 260 for 65portx10G) and we can also keep code simple.